#16541310 Users should be able to give a rating to an article

ericnyirimana opened the pull request:
Description
=======
Author heaven an application where writers can publish their articles. This feature permits other users to rate writers articles whether is a good reading or not.

Type of change
=======
N/A

How has it been tested
=======
- It is tested by POST ```api/v1/article/rate/<article-id>```

Checklist:
=======
N/A

Pivotal tracker story ID
=======
[165413107](https://www.pivotaltracker.com/n/projects/2326984/stories/165413107)

ericnyirimana committed:
Feat(rating): User should rate an article
- User rate other user article
- User can update their rates
- Wrote unit tests
- Documented api using swagger
[WIP #165413107]

ericnyirimana committed:
feat(rating): Changing updating status code
- change updating status code from 201 to 200 in controllers
- expect 200 status code when updating in tests
[Delivers #165413107]

AJAkimana reviewed (approved):
Good work

macphilips commented on helpers/articleRatevalidator.js:
```diff
@@ -0,0 +1,48 @@
+       * @param  {integer} user - Check the user
+       * @return {string} Validate the rate
+       */
+  static async validatearticleRated(article, user) {
```
```suggestion
  static async validateArticleRated(article, user) {
```

macphilips commented on middlewares/article.js:
```diff
@@ -0,0 +1,36 @@
+     * @return {string} Returns error or proceed the execution
+     * @static
+     */
+  static async checkRatedarticle(req, res, next) {
```
```suggestion
  static async checkRatedArticle(req, res, next) {
```

ericnyirimana committed:
feat(rating): changed article rate function name in helpers
- Changed validatearticleRated to validateSrticleRated in article helper
- Changed checkRatedarticle to checkRatedArticle in article middleware
[Delivers #165413107]
